### ISSUE TITLE
Soroban getEvents cursor is giving empty response

### ISSUE THREAD

**Issue by ella-quicknode** [stellar/soroban-rpc#313]
### What version are you using?

Node is implemented with `Soroban v21.5.1`

`getNetwork` response is: 

```
{
  "jsonrpc": "2.0",
  "id": 1,
  "result": {
    "friendbotUrl": "https://friendbot.stellar.org/",
    "passphrase": "Public Global Stellar Network ; September 2015",
    "protocolVersion": 21
  }
}
```

`getHealth` response is: 

```
{
  "jsonrpc": "2.0",
  "id": 8675309,
  "result": {
    "status": "healthy",
    "latestLedger": 53866455,
    "oldestLedger": 53849176,
    "ledgerRetentionWindow": 17280
  }
}
```

### What did you do?

When calling `getEvents`, without cursor, it worked fine. node returns data, but when using cursor in the params, it then returns empty response. 

Working payload without `cursor` : 

```
{
  "jsonrpc": "2.0",
  "id": 1,
  "method": "getEvents",
  "params": {
    "startLedger": 53849842,
    "pagination": {
      "limit": 3
    }
  }
}
```

Empty response when query with `cursor`, tried recent block within 24 hrs, the same behaviour 

```
{
  "jsonrpc": "2.0",
  "id": 8675309,
  "method": "getEvents",
  "params": {
    "pagination": {
      "limit": 3,
      "cursor": "0231283310285574144-0000000001"
    }
  }
}
```

### What did you expect to see?

It looked like the nodes aren't able to recognise cursor.  

If calling the same payload on this url: https://soroban-rpc.mainnet.stellar.gateway.fm/
data is returned. 

Is there config we missed out from the node?

**Comment by 2opremio** [comment]
@Shaptic did you get to make any progress on this?

**Comment by Shaptic** [comment]
Iirc my working theory was having to do with the 10k scan limit; I couldn't find a reasonable explanation otherwise

**Comment by Shaptic** [comment]
This is closed by v22.0.0
